src/aspire/source/relion.py

```diff
@@ -11,47 +10,48 @@
 from aspire.source import ImageSource
 from aspire.image import Image
 from aspire.io.starfile import StarFile
-from aspire.utils.filters import CTFFilter, PowerFilter
-from aspire.source.xform import FilterXform
-from aspire.estimation.noise import WhiteNoiseEstimator
+from aspire.utils.filters import CTFFilter
 
 logger = logging.getLogger(__name__)
 
 
 class RelionSource(ImageSource):
 
     @classmethod
-    def starfile2df(cls, filepath, data_folder=None, max_rows=None):
-        if data_folder is not None:
-            if not os.path.isabs(data_folder):
-                data_folder = os.path.join(os.path.dirname(filepath), data_folder)
-        else:
-            data_folder = os.path.dirname(filepath)
+    def starfile2df(cls, proj_folder, filepath, max_rows=None):
 
+        star_folder = proj_folder
+        if proj_folder is not None:
+            if not os.path.isabs(filepath):
+                star_folder = os.path.join(proj_folder, os.path.dirname(filepath))
+        else:
+            star_folder = os.path.dirname(filepath)
+        starfile_path = os.path.join(star_folder, os.path.basename(filepath))
         # Note: Valid Relion image "_data.star" files have to have their data in the first loop of the first block.
         # We thus index our StarFile class with [0][0].
-        df = StarFile(filepath)[0][0]
+        df = StarFile(starfile_path)[0][0]
         column_types = {name: cls.metadata_fields.get(name, str) for name in df.columns}
         df = df.astype(column_types)
 
+        # TODO: The statement need to be reimplemented in future version of pandas
         _index, df['__mrc_filename'] = df['_rlnImageName'].str.split('@', 1).str
         df['__mrc_index'] = pd.to_numeric(_index)
 
         # Adding a full-filepath field to the Dataframe helps us save time later
         # Note that os.path.join works as expected when the second argument is an absolute path itself
-        df['__mrc_filepath'] = df['__mrc_filename'].apply(lambda filename: os.path.join(data_folder, filename))
+        df['__mrc_filepath'] = df['__mrc_filename'].apply(lambda filename: os.path.join(proj_folder, filename))
 
         if max_rows is None:
             return df
         else:
             return df.iloc[:max_rows]
 
-    def __init__(self, filepath, data_folder=None, pixel_size=1, B=0, n_workers=-1, max_rows=None, memory=None):
+    def __init__(self, proj_folder, filepath, pixel_size=1, B=0, n_workers=-1, max_rows=None, memory=None):
```

Comment:
I don't quite see why the notion of a “project” should touch a `RelionSource`. What does it need to know other than where to find the MRC files?

Reply:
As I replied for a comment above,  I tend to set a uniform format of all modules including cov2d, cov3d and others. The `proj_folder` specifies the absolute path to the project folder with respect to all relative paths to the star files for different analysis such as denoising and 2D classification. Inside the star file, the paths for mrcs files will be relative to this `proj_folder` too. In this way, no matter where is star file, the RelionSource function can find the mrc files using the path of `proj_folder` and the entry in the star file. This will be more consistent with the Relion package. The `filepath` should include the relative path to project folder and input STAR file.

Reply:
Same as above. I reverted back to the original format using the data folder and created an new issue #141 for future discussion.